zerver/lib/upload.py

```diff
@@ -459,6 +493,25 @@ def ensure_medium_avatar_image(self, email):
         resized_medium = resize_avatar(image_data, MEDIUM_AVATAR_SIZE)
         write_local_file('avatars', file_path + '-medium.png', resized_medium)
 
+    def upload_emoji_image(self, emoji_file, emoji_file_name, user_profile):
+        # type: (File, Text, UserProfile) -> None
+        emoji_path = RealmEmoji.PATH_ID_TEMPLATE.format(
+            realm_id= user_profile.realm_id,
+            emoji_file_name=emoji_file_name
+        )
+
+        image_data = emoji_file.read()
+        write_local_file(
+            'avatars',
+            emoji_path,
+            image_data)
+
+    def get_emoji_url(self, emoji_file_name, realm_id):
+        # type: (Text, int) -> Text
+        return os.path.join(
+            u"/user_avatars",
+            RealmEmoji.PATH_ID_TEMPLATE.format(realm_id=realm_id, emoji_file_name=emoji_file_name))
```

Comment:
Yeah, I see.  Maybe worth checking what other products do with animated gifs?  I think it's possible we just need to figure out how to resize those, since they're going to be super large and thus benefit the most from resizing, but I could also imagine as a temporary workaround just reporting an error if someone uploads a too-large animated gif.